test/test-tools.go

```diff
@@ -177,3 +177,18 @@ func CountCounter(counter prometheus.Counter) int {
 	_ = m.Write(&iom)
 	return int(iom.Counter.GetValue())
 }
+
+func CountHistogramSamples(hist prometheus.Histogram) int {
+	ch := make(chan prometheus.Metric, 10)
+	hist.Collect(ch)
+	var m prometheus.Metric
+	select {
+	case <-time.After(time.Second):
+		fmt.Println("timed out collecting metrics")
```

Comment:
Should we treat this as a `panic` instead? Maybe return an `error` in addition to the `int`?

Presently `TestPerformValidation` has a check `if samples != 0` that it uses to determine if the test passed. It seems like it wouldn't be able to distinguish between a true `0` return and a passing test or a case where `CountHistogramSamples` began to timeout on collection. We would likely miss the `Printf` in CI output for this case and have a test that passes when it should fail.